### ISSUE TITLE
[1.49] Don't display "Approve" button in tokens list for native token

### ISSUE THREAD

**Issue by shoom3301** [cowprotocol/cowswap#3344]
**Description**

<img width="1005" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/2959f016-a67e-4d31-a1cc-5b9b72e6aadd">

**How to Reproduce**
1. Go to '/account/tokens'

**Expected behavior**
"Approve" button is not disabled for native token

**Comment by t0rbik** [comment]
Was looking at this and wonder if [nativeAndWrappedTokens.ts](https://github.com/cowprotocol/cowswap/blob/caddc9b59f16979807ca5790fbcc1ddfd5bc74a6/libs/common-const/src/nativeAndWrappedTokens.ts) should actually pass `isNative` set to `true` to `TokenWithLogo` constructor? 
Then in [TokensTableRow.txs](https://github.com/cowprotocol/cowswap/blob/caddc9b59f16979807ca5790fbcc1ddfd5bc74a6/apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx) you could just: 

```tsx
const displayApproveContent = useMemo(() => {
    if (tokenData.isNative) {
      return null;
    }
...
```

Other option is to do similar to some other places:

```tsx
...
const isNative = getIsNativeToken(tokenData)

  const displayApproveContent = useMemo(() => {
    if (isNative) {
      return null
    }
...
```

and then for etherscan link:

```html
...
<Cell>
        {!isNative &&
          <ExtLink href={getBlockExplorerUrl(chainId, 'token', tokenData.address)}>
            <TableButton>
              <SVG src={EtherscanImage} title="View token contract" description="View token contract" />
            </TableButton>
          </ExtLink>}
        {displayApproveContent}
</Cell>
...
```

In any case lmk if you want a PR as this Trivial indeed.

**Comment by elena-zh** [comment]
Fixed in #3786
